Python/Product/WFastCgi/wfastcgi.py

```diff
@@ -803,6 +803,13 @@ def main():
                     record.params['SCRIPT_NAME'] = ''
                     record.params['wsgi.script_name'] = wsgi_encode('')
 
+                # correct SCRIPT_NAME and PATH_INFO if we are told what our SCRIPT_NAME should be
+                if 'SCRIPT_NAME' in os.environ:
```

Comment:
For safety, I'd prefer to check that `PATH_INFO` starts with `SCRIPT_NAME` as part of this condition.